### DISCUSSION TITLE
Grid Filters: Nested Object filter

### DISCUSSION THREAD

**Discussion by nizarMathlouthi** [vikramlearning/blazorbootstrap#389] [General]
how to filter on an oject  complex column

exmple : Employe.Adress.PostalCode

**Comment by gvreddy04** [comment]
@nizarMathlouthi Thank you for trying **BlazorBootstrap** components. Can you share a sample mimicking your scenario? It will help me understand better.

**Reply by benholtsclaw** [reply]
I have this same issue with client side filtering. In a grid, I'm displaying a collection of Applicants. The applilcant's grade is displayed in a column. Displaying the property works fine.

```csharp
<GridColumn TItem="Applicant" HeaderText="Grade" PropertyName="Grade.Name" SortKeySelector="x => x.Grade.NumericalRepresentation">
            @context.Grade.Name
</GridColumn>
```

I have these classes. Applicant's FirstName and LastName fields are filterable without issue. Grade filter does not work.

```csharp
public class Applicant : IEquatable<Applicant>, IEqualityComparer<Applicant>, IComparable<Applicant>
{
    public int Id { get; set; }

    public string FirstName { get; set; }

    public string LastName { get; set; }

    public int GradeId { get; set; }

    public Grade Grade { get; set; }
}

public class Grade
{
    public int Id { get; set; }
    
    public int NumericalRepresentation { get; set; }

    public string Name { get; set; }

    public string Description { get; set; }

    public ICollection<Applicant> Applicants { get; set; }

    public override string ToString()
    {
        return this.Name;
    }
}
```

The Grade filter shows no equality operators and does nothing when you type in the box.

![image](https://github.com/vikramlearning/blazorbootstrap/assets/723813/fcb74fb2-e564-43eb-a051-7b959dee8926)

**Comment by nizarMathlouthi** [comment]
`<Grid TItem="ScheduleDetailsDto" class="table table-hover table-bordered table-striped"
      DataProvider="EmployeesDataProvider"
      Responsive=true
      AllowFiltering=true
      AllowSorting=true
      EmptyText="Aucune données..."
      ItemsPerPageText="Items per page"
      PaginationItemsTextFormat="{0} - {1} of {2} items"
      PageSizeSelectorItems="new int[] { 10, 20, 50 }">
    <GridColumn TItem="ScheduleDetailsDto" HeaderText="TacheId" PropertyName="Id" Filterable=false Sortable=false>
        @context.Id
    </GridColumn>
    <GridColumn TItem="ScheduleDetailsDto" HeaderText="Code operation" Filterable=false SortKeySelector ="item => item.Operation.OperationCode">
        @context.Operation.OperationCode
    </GridColumn>
    <GridColumn TItem="ScheduleDetailsDto" HeaderText="Nom operation" Filterable=false>
        @context.Operation.Name
    </GridColumn>
    <GridColumn TItem="ScheduleDetailsDto"  HeaderText="Date création" PropertyName="CreatedAt" SortKeySelector="item => item.CreatedAt">
        @context.CreatedAt.ToShortDateString()
    </GridColumn>
    <GridColumn TItem="ScheduleDetailsDto" HeaderText="Complete" PropertyName="Complete" Filterable=false Sortable=false>
        @context.Complete
    </GridColumn>
    <GridColumn TItem="ScheduleDetailsDto" HeaderText="Export" Filterable=false Sortable=false>
        <ChildContent>
            <div class="row justify-content-center">
                <div class="col-sm-auto">
                    <Button Disabled="@(!context.Complete)" Color="ButtonColor.Primary" @onclick=@(async ()=>await DownloadFileFromStream(context.Id))>Export</Button>
                </div>
            </div>
        </ChildContent>
    </GridColumn>
</Grid>
<Toasts class="p-3" AutoHide="true" Delay="3000" Messages="messages" Placement="ToastsPlacement.BottomRight" />`

**Reply by nizarMathlouthi** [reply]
complex object
`
public partial class ScheduleDetailsDto:ScheduleDto
{
    public SettingDto? Setting { get; set; }=new SettingDto();
    public OperationDto? Operation { get; set; }=new OperationDto();
}`

**Comment by benholtsclaw** [comment]
For anyone else who, like me, came to the discussion forum first seeking out this issue -- 
It's an actual issue. See https://github.com/vikramlearning/blazorbootstrap/issues/542
